src/main/java/com/rybka/service/exchange/BankAggregatorService.java

```diff
@@ -0,0 +1,73 @@
+package com.rybka.service.exchange;
+
+import com.rybka.constant.BankAggregatorConstants;
+import com.rybka.constant.Messages;
+import com.rybka.exception.AggregatorConnectionException;
+import com.rybka.exception.ParameterParseException;
+import com.rybka.model.BankData;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Element;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@Data
+@RequiredArgsConstructor
+public class BankAggregatorService {
+
+    private final HttpClient httpClient;
+    private final String DATE_PATTERN = "yyyy-M-dd hh:mm";
+
+    public List<BankData> loadExchangeRatesPageFor(String currency) {
+        try {
+            var request = HttpRequest.newBuilder()
+                    .uri(URI.create((String.format(BankAggregatorConstants.BANK_AGGREGATOR_URL, currency))))
+                    .build();
+            var response = httpClient.send(request,
+                    HttpResponse.BodyHandlers.ofString());
```

Comment:
add separate method for request and response.

Reply:
fixed